executor/include/executor/default/omp_executor.hpp

```diff
@@ -54,11 +68,14 @@ struct omp_executor {
   }
 
   template <typename F>
-  void bulk_execute(F&& f, shape_type n) const {
+  void bulk_execute(F&& f, const shape_type n) const {
```

Comment:
If OpenMP is unavailable, then should this function even exist? :thinking:

Reply:
I added static asserts to his. It will be easier to understand the error this way than just making the function non-existent case OMP is unavailable.

Reply:
The assumption is that it will only trigger compilation failures if the user explicitly tries to use it right? Is best_fit capable of triggering the instantiation in case OMP is not available?

Reply:
In 87580f6 I added this.
So if OMP is not available the best fit executor will just skip over OMP executor and it will not trigger a compilation failed and will also never be selected.